travis-ci/managers/ubuntu-build-check.sh

```diff
@@ -0,0 +1,102 @@
+#!/bin/bash
+
+set -e
+
+info() { echo -e "\033[33;1m$1\033[0m"; }
+error() { echo >&2 -e "\033[31;1m$1\033[0m"; }
+success() { echo >&2 -e "\033[32;1m$1\033[0m"; }
+
+ARGS=(
+    "--optimization=0"
+    "--optimization=2"
+    "--optimization=3"
+    "--optimization=s"
+    "-Db_lto=true"
+    "-Db_ndebug=true"
+)
+PACKAGES=(
+    cryptsetup-bin
+    expect
+    fdisk
+    gettext
+    iptables-dev
+    iputils-ping
+    isc-dhcp-client
+    itstool
+    kbd
+    libblkid-dev
+    libcap-dev
+    libcurl4-gnutls-dev
+    libfdisk-dev
+    libgpg-error-dev
+    liblz4-dev
+    liblzma-dev
+    libmicrohttpd-dev
+    libmount-dev
+    libp11-kit-dev
+    libpwquality-dev
+    libqrencode-dev
+    libssl-dev
+    libxkbcommon-dev
+    libzstd-dev
+    mount
+    net-tools
+    ninja-build
+    perl
+    python-lxml
+    python3-evdev
+    python3-lxml
+    python3-pip
+    python3-pyparsing
+    python3-setuptools
+    quota
+    strace
+    unifont
+    util-linux
+    zstd
+)
+CC="${CC:-gcc}"
+CXX="${CXX:-g++}"
+AR="${AR:-""}"
+RELEASE="${1:-bionic}"
```

Comment:
Since we expect those variables to be always set I think it would be better to drop these altogether and either use something like `${CC:?}` when `meson` is run or add `set -u` and keep using just `$CC`. It would have helped to catch the issue mentioned in https://github.com/systemd/systemd/pull/16136#issuecomment-642328608.

Instead of `RELEASE` it should be possible to rely on `lsb_release -cs`.

Reply:
I went with `"CC="${CC:?}"` as it *explicitly* say we expect these variables to be set, but otherwise it works the same way as `set -e`.

And thanks a lot for the catch in https://github.com/systemd/systemd/pull/16136#issuecomment-642328608 _ I forgot to pass `-E` when I moved to `sudo`.